src/napari_omero/plugins/loaders.py

```diff
@@ -241,3 +241,143 @@ def get_lazy_big_plane(level_id, level_desc, z, c, t):
         pyramid.append(da.stack(t_stacks))
 
     return pyramid
+
+
+def load_rois(conn: BlitzGateway, image: ImageWrapper) -> list[LayerData]:
+    """Retrieve ROIs from an OMERO image and format them as a single Napari layer."""
+    roi_service = conn.getRoiService()
+    result = roi_service.findByImage(image.getId(), None)
+    img_id = image.getId()
+
+    # Lists to store properties for all shapes
+    all_coords = []
+    all_shape_types = []
+    all_comments = []
+    all_roi_ids = []
+    all_shape_ids = []
+    all_z_indices = []
+    all_t_indices = []
+
+    # Get Z and T dimension ranges for the image
+    size_z = range(image.getSizeZ())
+    size_t = range(image.getSizeT())
+
+    # Get physical pixel sizes; default to 1 if not defined
+    size_x = image.getPixelSizeX() or 1
+    size_y = image.getPixelSizeY() or 1
+    pixel_size_z = image.getPixelSizeZ() or 1
+
+    # Loop through all ROIs
+    for roi in result.rois:
+        roi_id = roi.getId().getValue()
+
+        # Loop through all shapes in each ROI
+        for shape in roi.copyShapes():
+            if shape is None:
+                continue  # Skip empty shapes
+
+            # Parse shape to Napari-compatible format
+            parsed = parse_omero_shape(shape)
+            if parsed is None:
+                continue  # unsupported shape
+
+            coords_2d, meta, _ = parsed
+            coords_2d = [[y, x] for y, x in coords_2d]
+
+            shape_id = shape.getId().getValue()
+            labels = shape.getTextValue()
```

Comment:
Is it possible to also get the color of the ROI on OMERO? Would be cool to also get that 1:1 from OMERO